Remove Alcatraz from Xcode skipped plugins list on install

guillaume-algis opened the pull request:
Related to #281, #282, #303 and probably a whole bunch of other issues.

This is very similar #282, but just removes Alcatraz from the skipped plugins list instead of un-blocking everything.
I also took the chance to add ["strict flags"](http://redsymbol.net/articles/unofficial-bash-strict-mode/) to the sh script, and echoing the current selected version of Xcode after install.

guillaume-algis commented:
Rebased this PR on top of master. Could you take a look? Thanks!

kattrali commented on Scripts/install.sh:
```diff
@@ -1,11 +1,26 @@
+XCODE_VERSION="$(xcrun xcodebuild -version | head -n1 | awk '{ print $2 }')"
+PLIST_PLUGINS_KEY="DVTPlugInManagerNonApplePlugIns-Xcode-${XCODE_VERSION}"
+BUNDLE_ID="com.mneorr.Alcatraz"
+TMP_FILE="/tmp/${BUNDLE_ID}.xcode-defaults"
```
Probably should add `$(random)` somewhere in here to avoid collisions

supermarin replied:
@kattrali not sure `random` is installed on OSX by default.

`mktemp` might work

kattrali replied:
oh right, `echo $RANDOM`

kattrali commented on Scripts/install.sh:
```diff
@@ -1,11 +1,26 @@
+# Remove Alcatraz from Xcode's skipped plugins list if needed
+defaults read -app Xcode "$PLIST_PLUGINS_KEY" > "$TMP_FILE"
+/usr/libexec/PlistBuddy -c "delete skipped:$BUNDLE_ID" "$TMP_FILE" > /dev/null 2>&1 && {
+	pgrep Xcode > /dev/null && { echo 'An instance of Xcode is currently running. Please close Xcode before installing Alcatraz.'; exit 1; }
```
Its easier to read this section if you spread everything after the curly brace onto separate lines, just like the outer enclosure.

kattrali commented:
LGTM, added inline comments.

supermarin commented on Scripts/install.sh:
```diff
@@ -1,11 +1,26 @@
+/usr/libexec/PlistBuddy -c "delete skipped:$BUNDLE_ID" "$TMP_FILE" > /dev/null 2>&1 && {
+	pgrep Xcode > /dev/null && { echo 'An instance of Xcode is currently running. Please close Xcode before installing Alcatraz.'; exit 1; }
+	defaults write -app Xcode "$PLIST_PLUGINS_KEY" "$(< $TMP_FILE)"
+	echo 'Alcatraz was removed from Xcode'\''s skipped plugins list. Next time you start Xcode select "Load Bundle" when prompted.'
```
If possible, it'd be appreciated if length could be <80

guillaume-algis commented:
I made the requested changes:
- Used `mktemp` instead of `$RANDOM` as it made the intent clearer;
- Replaced `"$(< $TMP_FILE)"` by `"$(cat "$TMP_FILE")"` as the `$(< )` construct is apparently not POSIX compliant;
- Made most of the script fit on 80 cols. The remaining lines are harder to compress, and would harm readability imo.
- Replaced tabs with spaces;

kattrali commented:
:+1: Awesome! Thanks @guillaume-algis, looks good.
